nautobot/dcim/models/devices.py

```diff
@@ -979,3 +1034,122 @@ def devices_sorted(self):
 
     def __str__(self):
         return self.name
+
+
+#
+# Software images
+#
+
+
+class SoftwareImageQuerySet(RestrictedQuerySet):
+    """Queryset for SoftwareImage objects."""
+
+    def get_for_object(self, obj):
+        """Return all SoftwareImage assigned to the given object."""
+        if not isinstance(obj, models.Model):
+            raise TypeError(f"{obj} is not an instance of Django Model class")
+        if isinstance(obj, Device):
+            qs = self.filter(software_version__platform__devices=obj)
+        elif isinstance(obj, InventoryItem):
+            qs = self.filter(software_version__platform__devices__inventory_items=obj)
+        else:
+            qs = self
+
+        return qs
+
+
+@extras_features(
+    "custom_links",
+    "custom_validators",
+    "export_templates",
+    "graphql",
+    "statuses",
+    "webhooks",
+)
+class SoftwareImage(PrimaryModel):
+    """A software image for a Device, Virtual Machine or Inventory Item."""
+
+    software_version = models.ForeignKey(
+        to="SoftwareVersion", on_delete=models.CASCADE, related_name="software_images", verbose_name="Software Version"
+    )
+    image_file_name = models.CharField(blank=False, max_length=255, verbose_name="Image File Name")
+    image_file_checksum = models.CharField(blank=True, max_length=256, verbose_name="Image File Checksum")
```

Comment:
what is the reason for `max_length` differ here?

Reply:
I brought this over from DLM. My guess is that since a sha512 checksum is 128 characters this was to future proof?